docs/4.examples/define-route.md

```diff
@@ -0,0 +1,331 @@
+---
+icon: ph:arrow-right
+---
+
+# Define Route
+
+> A structured approach to route definition with optional type safety, validation, and seamless h3 integration.
+
+The `defineRoute` function provides a consistent, type-safe way to define routes in h3. It's designed to be simple and compatible with existing h3 patterns while offering powerful validation capabilities.
+
+## Why defineRoute?
+
+In modern applications, you often need:
+
+- **Type safety** for route parameters and request bodies
+- **Consistent validation** across all endpoints
+- **Route metadata** for authentication, caching, rate limiting
+- **Middleware integration** for cross-cutting concerns
+- **Better developer experience** with IntelliSense and error checking
+- **Seamless integration** with schema libraries like Zod, Valibot, etc.
```

Comment:
All of above are achieved with normal `defineValidated` utility. `defineRoute` was proposed to allow modular/encapsulated registration.